source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -1,3 +1,447 @@
+   };
+
+.. class:: train_result
+
+   .. function:: train_result()
+
+      Creates a result of the training operation with the default attribute
+      values.
+
+
+   .. member:: pca::model model = pca::model()
+
+      The trained K-means model.
+
+      Getter
+         | ``const model& get_model() const``
+
+
+   .. member:: table means = table()
+
+      :math:`1 \times r` table that contains mean value for each feature.
+
+      Getter
+         | ``const table& get_means() const``
+
+
+   .. member:: table variances = table()
+
+      :math:`1 \times r` table that contains variance for each feature.
+
+      Getter
+         | ``const table& get_variances() const``
+
+
+   .. member:: table eigenvalues = table()
+
+      :math:`1 \times r` table that contains eigenvalue for each feature.
+
+      Getter
+         | ``const table& get_eigenvalues() const``
+
+
+Operation semantics
+~~~~~~~~~~~~~~~~~~~
+.. namespace:: onedal
+.. function:: template <typename Descriptor> \
+              pca::train_result train(const Descriptor& desc, \
+                                      const pca::train_input& input)
+
+   :tparam Descriptor: PCA algorithm descriptor :expr:`pca::desc`.
+
+   Preconditions
+      | :expr:`input.data.is_empty == false`
+      | :expr:`input.data.column_count >= desc.component_count`
+
+   Postconditions
+      | :expr:`result.means.row_count == 1`
+      | :expr:`result.means.column_count == desc.component_count`
+      | :expr:`result.variances.row_count == 1`
+      | :expr:`result.variances.column_count == desc.component_count`
+      | :expr:`result.eigenvalues.row_count == 1`
+      | :expr:`result.eigenvalues.column_count == desc.component_count`
+      | :expr:`result.model.eigenvectors.row_count == 1`
+      | :expr:`result.model.eigenvectors.column_count == desc.component_count`
+
+
+Inference :expr:`onedal::infer(...)`
+------------------------------------
+Input
+~~~~~
+::
+
+   class infer_input {
+   public:
+      infer_input();
+      infer_input(const model& m);
+      infer_input(const model& m, const table& data);
+
+      const model& get_model() const;
+      const table& get_data() const;
+
+      infer_input& set_model(const model&);
+      infer_input& set_data(const table&);
+   };
+
+.. namespace:: onedal::pca
+
+.. class:: infer_input
+
+   .. function:: infer_input()
+
+      Creates an input for the inference operation with the default attribute
+      values.
+
+
+   .. function:: infer_input(const pca::model& model)
+
+      Creates an input for the inference operation with the given :expr:`model`,
+      the other attributes get default values.
+
+
+   .. function:: infer_input(const pca::model& model, const table& data)
+
+      Creates an input for the inference operation with the given :expr:`model`
+      and :expr:`data`.
+
+
+   .. member:: table data = table()
+
+      :math:`n \times p` table with the data to be projected to the :math:`r`
```

Comment:
it looks another candidate to add the notation and explain n rows and p columns explicitly to avoid different intepretation

Reply:
Agree